docker/fetch_normalization_data.py

```diff
@@ -0,0 +1,197 @@
+    dbname = os.environ.get('resource.postgresql.database_name', 'breakpad')
+
+    local_config = {
+        'database_hostname': host,
+        'database_port': port,
+        'database_username': username,
+        'database_password': password,
+        'database_name': dbname
+    }
+    dsn = (
+        "host=%(database_hostname)s "
+        "dbname=%(database_name)s "
+        "port=%(database_port)s "
+        "user=%(database_username)s "
+        "password=%(database_password)s"
+    ) % local_config
+    return psycopg2.connect(dsn)
+
+
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--products', default='Firefox',
+        help='comma separated set of products to fetch ADI for'
+    )
+    parser.add_argument(
+        '--start_date', default='LASTWEEK', help='start date (mm/dd/yyyy)'
+    )
+    parser.add_argument(
+        '--end_date', default='TODAY', help='end date (mm/dd/yyyy)'
+    )
+    parser.add_argument(
+        '--endpoint', default='https://crash-stats.mozilla.com/api/ADI/',
+        help='endpoint for ADI API'
+    )
+    args = parser.parse_args()
+
+    today = datetime.datetime.now()
+    last_week = datetime.datetime.now() - datetime.timedelta(days=7)
+
+    start_date = args.start_date
+    if start_date == 'LASTWEEK':
+        start_date = last_week.strftime('%m/%d/%Y')
+
+    end_date = args.end_date
+    if end_date == 'TODAY':
+        end_date = today.strftime('%m/%d/%Y')
+
+    if start_date > end_date:
+        print('Start date (%s) cannot be after end date (%s). Exiting.' % (start_date, end_date))
+        return 1
+
+    conn = get_connection()
+
+    print('Fetching normalization data for %s from %s to %s' % (
+        args.products, start_date, end_date)
+    )
+
+    for product in args.products.split(','):
+        # Fetch versions from the product_versions table that we should pull data for
+        versions = fetch_versions(conn, product)
+
+        for version in versions:
+            for platform in PLATFORMS:
+                # For each (product, version, platform) combo, fetch adu data and insert it
+                print('Fetching data for (%s, %s, %s)...' % (product, version, platform))
+
+                adu_data = fetch_data_from_api(
+                    endpoint=args.endpoint,
+                    product=product,
+                    version=version,
+                    platform=platform,
+                    start_date=start_date,
+                    end_date=end_date
+                )
+
+                for hit in adu_data:
+                    insert_adu(
+                        conn=conn,
+                        adu_count=hit['adi_count'],
+                        adu_date=hit['date'],
+                        product_name=product,
+                        os_name=platform,
+                        version_string=hit['version'],
+                    )
```

Comment:
Perhaps it's nitpickery but all of this can be simplified. The `/api/ADI/` uses an `OR` clause on multiple platforms and multiple versions (and multiple products too). 
So to avoid having to do `len(products) x len(product_versions) x len(platforms)` number of network requests you can just do:

```python
for product in args.products.split(','):
    adi_data = fetch_data_from_api(
        product=product,
        versions=fetch_versions(conn, product),
        platforms=PLATFORMS,
        start_date=start_date,
        end_date=end_date
    )
    for hit in adi_data:
         insert_adi(...)
```